src/main/java/me/zeroeightsix/kami/util/astaralg/Astarpathfinder.java

```diff
@@ -0,0 +1,650 @@
+                    BaritoneAPI.getProvider().getPrimaryBaritone().getCustomGoalProcess().setGoalAndPath(new GoalXZ(stuckPos.getX(), stuckPos.getZ()));
+                }
+            }
+        } else {
+            if (!BlockUtils.checkForLiquid()) {
+                // if blockstuck let vanilla mechanics run (i.e dont cancel the event) seems to work fine
+                if (!blockstuck)
+                    elytraFlight.landing(currentEvent);
+            } else {
+                currentDirection = direction + 4 % 8;
+                pathfind();
+            }
+        }
+//        the plan was to do something like this:
+//        switch (landstate) {
+//            case FLY_TO_OBSTACLE:
+//                //TODO: Fly to the obstacle
+//                break;
+//            case LAND:
+//                //TODO:  find save landing spot and fly there
+//                break;
+//        }
+    }
+
+
+    private static void pathfind() {
+        mc.timer.tickLength = 50;
+        mc.player.setSprinting(false);
+        holdinair = false;
+        currentplayerBlockpos = getPlayerBlockpos();
+        calcPath(true);
+        removeAllBeforeClosest();
+        if (!checkNextChunkLoaded()) {
+            holdinair = true;
+        } else {
+            if (checkStuck() || mc.player.onGround || (!mc.player.isElytraFlying())) {
+                lastStuckCheckPos = new Vec3d(0, 0, 0);
+                currentStuckCheckPos = new Vec3d(0, 0, 0);
+                path = null;
+                stuckPos = getPlayerBlockpos();
+                state = State.LAND_SAVE;
+                nopathfound = 0;
+                elytraFlight.controlMode(currentEvent);
+                return;
+            }
+        }
+        float yawDeg = 0;
+        if (path != null && path.size() > 1) {
+            yawDeg = (float) EntityUtils.calculateLookAt(path.get(Math.min(path.size() - 1, 1)).pos.getX() + 0.5, mc.player.posY, path.get(Math.min(path.size() - 1, 1)).pos.getZ() + 0.5, mc.player)[0];
+        }
+        currentyaw = yawDeg;
+        elytraFlight.controlMode(currentEvent);
+
+
+    }
+
+    private static boolean checkNextChunkLoaded() {
+        return mc.world.getChunk(getPlayerBlockpos()).isLoaded() &&
+                mc.world.getChunk(getPlayerBlockpos().add(0, 0, 16)).isLoaded() &&
+                mc.world.getChunk(getPlayerBlockpos().add(0, 0, -16)).isLoaded() &&
+                mc.world.getChunk(getPlayerBlockpos().add(16, 0, 0)).isLoaded() &&
+                mc.world.getChunk(getPlayerBlockpos().add(-16, 0, 0)).isLoaded() &&
+                mc.world.getChunk(getPlayerBlockpos().add(16, 0, 16)).isLoaded() &&
+                mc.world.getChunk(getPlayerBlockpos().add(-16, 0, 16)).isLoaded() &&
+                mc.world.getChunk(getPlayerBlockpos().add(16, 0, -16)).isLoaded() &&
+                mc.world.getChunk(getPlayerBlockpos().add(-16, 0, -16)).isLoaded();
```

Comment:
Oh, this is smart. I was struggling to do #665 because I was testing the coordinates of chunks and it didn't work because coords were always 0, I could've just used relative pos :p

Reply:
tbh you could probably go further than that because you dont need to care for the chunks to the left and right of you if you just go forward but it works like this and accounting for direction would prob save less than a second in most cases.